core/src/main/java/de/danoeh/antennapod/core/service/FeedMediaSizeService.java

```diff
@@ -0,0 +1,57 @@
+package de.danoeh.antennapod.core.service;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.util.Log;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+
+import de.danoeh.antennapod.core.event.FeedMediaEvent;
+import de.danoeh.antennapod.core.feed.FeedMedia;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.DBWriter;
+import de.danoeh.antennapod.core.util.NetworkUtils;
+import de.greenrobot.event.EventBus;
+
+public class FeedMediaSizeService extends IntentService {
+
+    private final static String TAG = "FeedMediaSizeService";
+
+    public FeedMediaSizeService() {
+        super(TAG);
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+        Log.d(TAG, "onHandleIntent()");
+        if(false == NetworkUtils.networkAvailable(this)) {
+            return;
+        }
+        List<FeedMedia> list = DBReader.getFeedMediaUnknownSize(this);
+        for (FeedMedia media : list) {
+            if(false == NetworkUtils.networkAvailable(this)) {
+                return;
+            }
+            long size = Integer.MIN_VALUE;
+            Log.d(TAG, media.getDownload_url());
+            try {
+                URL url = new URL(media.getDownload_url());
+                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+                conn.setRequestProperty( "Accept-Encoding", "" );
+                conn.setRequestMethod("HEAD");
+                size = conn.getContentLength();
+                conn.disconnect();
```

Comment:
not sure how big of a deal this is, but since we can throw an exception that isn't about to crash the app, we should probably put conn.disconnect() in a 'finally' block to make sure it gets closed (assuming it's non-null).